test/generators/shop_model_generator_test.rb

```diff
@@ -29,6 +33,13 @@ class ShopModelGeneratorTest < Rails::Generators::TestCase
     end
   end
 
+  test "injects into application controller" do
+    run_generator
+    assert_file "app/controllers/application_controller.rb" do |controller|
+      assert_match "def shop", controller
+    end
```

Comment:
I get nothing for *shop* in any of the controllers while session[:shopify] is set. I ran the generators and am fiddling with a setup that comes pretty much straight from the current code at Github. I think I need to add the instantiation of shop in the application controller manually as per the point of this discussion...